### PR TITLE
Story.bar 12 - red button widget

### PR DESCRIPTION
**Pull request by stefanomusicman**
# Description of the problem
created a reusable red button that can be disabled, take any text and can carry out action

## How does it solve the problem
easily reuse the button anywhere in the app

# References (also the Jira ticket)
BAR-12

# Checklist:

- [ ] I re-reviewed the Jira ticket now to make sure 
- [ ] My code follows the style guidelines of this project & of Dart
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have wrote some documentation if I changed a big behavior of the code
- [ ] My changes generate no new warnings/and if they do, I described them to the team
- [ ] I have added tests that prove my fix is effective or that my feature works if I worked on a logic class/function
- [ ] New and existing unit and integration tests pass locally with my changes

### CODE REVIEW

File: lib/ui/buttons/red_button.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/material.dart';
+import '../../globals/theme/app_themes.dart';
+
+class RedButton extends StatefulWidget {
+  // title to be passed in
```
Comment (danielemonaca) [inline lib/ui/buttons/red_button.dart:5]:
```suggestion
```

I don't think these comments are very helpful

File: lib/ui/buttons/red_button.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  final String title;
+
+  // boolean value to pass in to activate/deactivate button
```
Comment (danielemonaca) [inline lib/ui/buttons/red_button.dart:8]:
```suggestion
```

Same here

File: lib/ui/buttons/red_button.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  final bool isActivated;
+
+  // function to be passed in to carry out desired action when button is clicked
```
Comment (danielemonaca) [inline lib/ui/buttons/red_button.dart:11]:
```suggestion
```

Same here

File: lib/ui/buttons/red_button.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+class _RedButtonState extends State<RedButton> {
+  late bool isButtonActive = widget.isActivated;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      onPressed: isButtonActive ? widget.action : null,
+      style: ElevatedButton.styleFrom(
```
Comment (danielemonaca) [inline lib/ui/buttons/red_button.dart:36]: is it possible that you forgot to give it a shadow?

image

Comment (stefanomusicman) [reply lib/ui/buttons/red_button.dart:36]: ya i didnt really think there was a drop shadow, i forgot to check, ill put it in now

**Comment by danielemonaca** [comment]
Can you also put a screenshot of how UI widgets look in the PR description from now on? I think it would be nice

**Comment by stefanomusicman** [comment]
yes will do

**Comment by stefanomusicman** [comment]
button

**Comment by danielemonaca** [comment]
Looks amazing!!

**Review by danielemonaca — APPROVED** [review]

**Pull request merged**